R/archivePackages.R

```diff
@@ -10,58 +10,84 @@
 ##' defaults to the value of the \dQuote{dratRepo} option with
 ##' \dQuote{"~/git/drat"} as fallback
 ##' @param type Character variable for the type of repository, so far
-##'  \dQuote{source}, \dQuote{win.binary}, \dQuote{mac.binary}, 
+##'  \dQuote{source}, \dQuote{win.binary}, \dQuote{mac.binary},
 ##'  \dQuote{mac.binary.mavericks} or \dQuote{mac.binary.el-capitan}
 ##' @param pkg Optional character variable specifying a package name(s), whose
-##' older versions should be archived. If missing (the default), archiving is 
+##' older versions should be archived. If missing (the default), archiving is
 ##' performed on all packages.
-##' @param version R version information in the format \code{X.Y} or 
+##' @param version R version information in the format \code{X.Y} or
 ##'   \code{X.Y.Z}. Only used, if archiving binary packages.
-##'   (default: \code{version = getRversion()})
+##'   By default all R versions found in the respective CRAN-like directory
+##'   for the chosen `type` are checked.
 ##' @examples
 ##' \dontrun{
 ##'   archivePackages()   # archive all older package versions
 ##'   archivePackages(pkg = "drat")  # archive older copies of just one package
 ##' }
 ##' @author Thomas J. Leeper
 archivePackages <- function(repopath = getOption("dratRepo", "~/git/drat"),
-                            type = "source", 
-                            pkg,
-                            version = getRversion()) {
+                            type = c(
+                                "source", "mac.binary", "mac.binary.el-capitan",
+                                "mac.binary.mavericks", "win.binary"
+                            ),
+                            pkg = NULL,
+                            version = NULL) {
```

Comment:
The suggested change is now simpler and more focussed.  One thing some people have learned to not be cavalier about is the change if existing interfaces.   `pkg` and `version` defaulted to "possibly empty" with a test for `is.missing`.  Having `NULL` is similar but not the same, and it *is* different.   There may be users out there with an existing script calling this (I am actually not entirely sure if/how anybody uses `archivePackages()` but I cannot know).  The use case of 'called with no argument' is preserved so I guess I can call that good enough.